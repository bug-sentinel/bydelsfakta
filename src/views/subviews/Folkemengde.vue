<template>
  <div class="main-container">
    <div class="main-container__cards">
      <h2 class="section-heading">Befolkningsutvikling og fremskriving</h2>
      <graph-card
        v-for="(card, index) in settings.fremskriving"
        :key="`folkemengde-fremskriving-${index}`"
        :settings="card"
      />
      <graph-card
        v-for="(card, index) in settings.utvikling"
        :key="`folkemengde-utvikling-${index}`"
        :settings="card"
      />
      <h2 class="section-heading">NÃ¸kkeltall for delbydelene</h2>
      <graph-card
        v-for="(card, index) in settings.nokkeltall"
        :key="`folkemengde-nokkeltall-${index}`"
        :settings="card"
      />
    </div>
    <div class="main-container__map">
      <h3>Kart</h3>
    </div>
  </div>
</template>

<script>
import GraphCard from '../../components/GraphCard.vue';

export default {
  name: 'Folkemengde',
  components: { GraphCard },
  props: {
    settings: {
      type: Object,
      required: true,
    },
  },
};
</script>

<style scoped>
</style>
