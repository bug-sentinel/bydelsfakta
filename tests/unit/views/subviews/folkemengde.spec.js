import { mount, createLocalVue } from '@vue/test-utils';
import VueResize from 'vue-resize';
import Folkemengde from '../../../../src/views/subviews/Folkemengde.vue';
import page from '../../../../src/config/pages';
import clickOutside from '../../../../src/directives/clickOutside';

describe('Folkemengde', () => {
  let wrapper = null;

  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(VueResize);
    localVue.directive('click-outside', clickOutside);
    wrapper = mount(Folkemengde, {
      propsData: {
        settings: page['folkemengde'],
      },
      localVue,
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('renders Folkemengde component correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
