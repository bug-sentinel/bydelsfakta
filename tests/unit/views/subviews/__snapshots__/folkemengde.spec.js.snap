// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folkemengde renders Folkemengde component correctly 1`] = `
<div
  class="main-container"
>
  <div
    class="main-container__cards"
  >
    <h2
      class="section-heading"
    >
      Befolkningsutvikling og fremskriving
    </h2>
     
    <div
      class="card-container large"
    >
      <div
        class="card"
      >
        <div
          class="tabs"
          role="tablist"
        >
          <button
            aria-label="Historisk"
            aria-selected="[object Object]"
            class="tabs__button active"
            role="tab"
          >
            Historisk
          </button>
          <button
            aria-label="Historisk (prosent)"
            aria-selected="[object Object]"
            class="tabs__button"
            role="tab"
          >
            Historisk (prosent)
          </button>
           
          <div
            class="context-menu"
            role="menu"
          >
            <button
              aria-haspopup="true"
              aria-label="åpne meny"
              class="context-menu__button"
            >
              <i
                class="material-icons"
              >
                menu
              </i>
            </button>
             
            <!---->
          </div>
        </div>
         
        <div
          class="graph__container"
        >
          <svg
            class="graph__svg"
            style="font-family: OsloSans;"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <text
              class="heading"
              font-size="14"
              font-weight="bold"
              y="14"
            />
            <g
              class="sidebar"
            >
              <g
                class="section-year"
              >
                <rect
                  class="section-year__bg"
                  fill="#292858"
                  height="40"
                  rx="4"
                  width="88"
                />
                <text
                  class="section-year__text"
                  fill="white"
                  font-size="26"
                  font-weight="700"
                  text-anchor="middle"
                  x="44"
                  y="30"
                />
              </g>
              <g
                class="section-population"
                transform="translate(0, 70)"
              >
                <rect
                  class="divider"
                  fill="#292858"
                  height="2"
                  width="250"
                />
                <text
                  class="label"
                  fill="#292858"
                  font-size="16"
                  font-weight="400"
                  y="26"
                >
                  Befolkning
                </text>
                <text
                  class="value"
                  fill="#292858"
                  font-size="16"
                  font-weight="700"
                  text-anchor="end"
                  x="250"
                  y="26"
                >
                  70
                </text>
              </g>
              <g
                class="section-change"
                transform="translate(0, 140)"
              >
                <rect
                  class="divider"
                  fill="#292858"
                  height="2"
                  width="250"
                />
                <text
                  class="label"
                  fill="#292858"
                  font-size="16"
                  font-weight="400"
                  y="26"
                >
                  Befolkningsvekst
                </text>
                <text
                  class="value"
                  fill="#292858"
                  font-size="16"
                  font-weight="700"
                  text-anchor="end"
                  x="250"
                  y="26"
                >
                  70
                </text>
                <rect
                  class="bar"
                />
              </g>
            </g>
            <g
              class="upper"
            >
              <g
                class="axis x"
              />
              <g
                class="axis y"
              />
              <path
                class="projection"
                fill="#6EE9FF"
              />
              <path
                class="expected"
                fill="none"
                stroke="#292858"
              />
              <path
                class="valueLine"
                fill="none"
                stroke="#292858"
                stroke-width="3"
              />
              <line
                class="currentDate"
                opacity="0.7"
                stroke="#292858"
                stroke-width="1"
                style="stroke-dasharray: 3,3;"
              />
              <text
                class="populationTooltip"
                font-size="12"
                text-anchor="end"
                transform="translate(-10, -3)"
              />
              <line
                class="highlightLine"
                stroke="#F8C66B"
                stroke-opacity="0.1"
                stroke-width="5"
              />
              <line
                class="highlightLine"
                stroke="#F8C66B"
                stroke-width="2"
              />
              <g
                class="highlightContainer"
              >
                <rect
                  class="highlightDateBox"
                  fill="#F8C66B"
                  height="26"
                  transform="translate(-24, 0)"
                  width="48"
                />
                <text
                  class="highlightText"
                  fill="#292858"
                  font-size="12"
                  font-weight="700"
                  text-anchor="middle"
                  y="18"
                />
              </g>
            </g>
            <g
              class="lower"
            >
              <g
                class="bars"
              />
              <line
                class="zeroline"
                stroke="#292858"
                stroke-width="1"
              />
              <g
                class="axis y"
              />
              <text
                class="sectionHeading"
                fill="#292858"
                font-weight="500"
              >
                Befolkningsendring per år
              </text>
            </g>
            <g
              class="triggers"
            />
            <g
              class="sourceGroup"
              fill="#292858"
              opacity="0"
            >
              <text
                class="source-label"
                font-size="10"
              >
                Kilde: 
              </text>
              <text
                class="source"
                font-size="10"
              />
            </g>
          </svg>
           
          <div
            class="resize-observer"
            data-v-b329ee4c=""
            tabindex="-1"
          >
            <object
              aria-hidden="true"
              data="about:blank"
              tabindex="-1"
              type="text/html"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="card-container large"
    >
      <div
        class="card"
      >
        <div
          class="tabs"
          role="tablist"
        >
          <button
            aria-label="Historisk"
            aria-selected="[object Object]"
            class="tabs__button active"
            role="tab"
          >
            Historisk
          </button>
          <button
            aria-label="Historisk (prosent)"
            aria-selected="[object Object]"
            class="tabs__button"
            role="tab"
          >
            Historisk (prosent)
          </button>
           
          <div
            class="context-menu"
            role="menu"
          >
            <button
              aria-haspopup="true"
              aria-label="åpne meny"
              class="context-menu__button"
            >
              <i
                class="material-icons"
              >
                menu
              </i>
            </button>
             
            <!---->
          </div>
        </div>
         
        <div
          class="graph__container"
        >
          <svg
            class="graph__svg"
            style="font-family: OsloSans;"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <text
              class="heading"
              font-size="14"
              font-weight="bold"
              y="14"
            />
            <g
              class="canvas"
              transform="translate(60, 50)"
            >
              <g
                class="axis x"
              />
              <g
                class="axis y"
              />
              <g
                class="axis x2"
              />
              <g
                class="axis y2"
              />
              <rect
                class="bg"
                fill="white"
              />
            </g>
            <g
              class="infobox"
            >
              <g
                class="infobox__body"
              >
                <rect
                  class="background"
                />
                <g
                  class="infobox__content"
                >
                  <text
                    class="infobox__heading"
                  />
                  <rect
                    class="infobox__rule"
                  />
                  <g
                    class="infobox__table"
                  />
                </g>
              </g>
              <g
                class="infobox__head"
              >
                <rect
                  class="background"
                />
                <text />
              </g>
            </g>
            <g
              class="sourceGroup"
              fill="#292858"
              opacity="0"
            >
              <text
                class="source-label"
                font-size="10"
              >
                Kilde: 
              </text>
              <text
                class="source"
                font-size="10"
              />
            </g>
          </svg>
           
          <div
            class="resize-observer"
            data-v-b329ee4c=""
            tabindex="-1"
          >
            <object
              aria-hidden="true"
              data="about:blank"
              tabindex="-1"
              type="text/html"
            />
          </div>
        </div>
      </div>
    </div>
     
    <h2
      class="section-heading"
    >
      Nøkkeltall for delbydelene
    </h2>
     
    <div
      class="card-container large"
    >
      <div
        class="card"
      >
        <div
          class="tabs"
          role="tablist"
        >
          <button
            aria-label="Nøkkeltall"
            aria-selected="[object Object]"
            class="tabs__button active"
            role="tab"
          >
            Nøkkeltall
          </button>
           
          <div
            class="context-menu"
            role="menu"
          >
            <button
              aria-haspopup="true"
              aria-label="åpne meny"
              class="context-menu__button"
            >
              <i
                class="material-icons"
              >
                menu
              </i>
            </button>
             
            <!---->
          </div>
        </div>
         
        <div
          class="graph__container"
        >
          <svg
            class="graph__svg"
            style="font-family: OsloSans;"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <text
              class="heading"
              font-size="14"
              font-weight="bold"
              y="14"
            />
            <g
              class="canvas"
              transform="translate(180, 90)"
            >
              <g
                class="axis x"
              />
              <g
                class="axis y"
              />
              <g
                class="axis x2"
              />
              <g
                class="axis y2"
              />
            </g>
            <g
              class="sourceGroup"
              fill="#292858"
              opacity="0"
            >
              <text
                class="source-label"
                font-size="10"
              >
                Kilde: 
              </text>
              <text
                class="source"
                font-size="10"
              />
            </g>
          </svg>
           
          <div
            class="resize-observer"
            data-v-b329ee4c=""
            tabindex="-1"
          >
            <object
              aria-hidden="true"
              data="about:blank"
              tabindex="-1"
              type="text/html"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="main-container__map"
  >
    <h3>
      Kart
    </h3>
  </div>
</div>
`;
